bolts-tasks/src/main/java/bolts/Task.java

```diff
@@ -115,6 +116,30 @@ public TResult getResult() {
   }
 
   /**
+   * @return The result of the task.
+   */
+  public TResult waitAndGetResult() throws InterruptedException, TimeoutException {
+    return waitAndGetResult(0, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * @return The result of the task.
+   */
+  public TResult waitAndGetResult(long duration, TimeUnit timeUnit) throws InterruptedException, TimeoutException {
+    synchronized (lock) {
+      if (!waitForCompletion(duration, timeUnit))
+        throw new TimeoutException();
+      if (isFaulted()) {
+        throw new RuntimeException(getError());
+      }
+      if (isCancelled()) {
+        throw new TaskCanceledException(this);
```

Comment:
Might be better to throw a java.util.concurrent.CancellationException to be consistent with the other cancellation exceptions